### PR TITLE
DataGrid - Add getVisibleColumnIndex method

### PR DESCRIPTION
**Pull request by igormalzew**

### CODE REVIEW

File: js/ui/grid_core/ui.grid_core.columns_controller.js [hunk 1]
```diff
@@ -2864,6 +2864,26 @@ module.exports = {
...
                 },
+
+                /**
+                 * @name GridBaseMethods.getVisibleColumnIndex
+                 * @publicName getVisibleColumnIndex(id)
+                 * @param1 id:number|string
+                 * @return number
+                 */
+                /**
+                 * @name dxTreeListMethods.getVisibleColumnIndex
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.columns_controller.js]: GridBaseMethods are base for dxDataGrid and dxTreeList. So the doc comments for dxTreeListMethods.getVisibleColumnIndex can be removed

File: testing/tests/DevExpress.ui.widgets.dataGrid/columnsController.tests.js [hunk 2]
```diff
@@ -7396,6 +7396,54 @@ QUnit.test("getVisibleColumns with rowIndex and grouped columns", function(asser
...
+                caption: "Band Column 1", columns: [
+                    { dataField: "TestField2", caption: "Column 2" },
+                    { dataField: "TestField3", caption: "Column 3" }
+                ]
+            }
+        ]
+    });
+
+    // assert
+    assert.ok(this.columnsController.isInitialized());
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets.dataGrid/columnsController.tests.js]: This assert is not necessary

File: testing/tests/DevExpress.ui.widgets.dataGrid/columnsController.tests.js [hunk 3]
```diff
@@ -7396,6 +7396,54 @@ QUnit.test("getVisibleColumns with rowIndex and grouped columns", function(asser
...
+
+    assert.equal(this.columnsController.getVisibleColumnIndex("name1"), 2, "by name");
+    assert.equal(this.columnsController.getVisibleColumnIndex(3), 3, "by column index");
+    assert.equal(this.columnsController.getVisibleColumnIndex("Column 1"), 0, "first match if not unique caption");
+    assert.equal(this.columnsController.getVisibleColumnIndex("dataField1"), 3, "first match if not unique dataField");
+    assert.equal(this.columnsController.getVisibleColumnIndex("foo"), -1, "non existing key");
```
Comment (vconst) [inline testing/tests/DevExpress.ui.widgets.dataGrid/columnsController.tests.js:7440]: Add assert with incorrect number index

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/grid_core/ui.grid_core.columns_controller.js [hunk 4]
```diff
@@ -2864,6 +2864,20 @@ module.exports = {
                     }
                     return -1;
                 },
+
+                /**
+                 * @name GridBaseMethods.getVisibleColumnIndex
```
Comment (AntonSermyazhko) [inline js/ui/grid_core/ui.grid_core.columns_controller.js:2869]: Will this method work if: { scrolling: { columnRenderingMode="virtual" } } (for virtual columns) ?
If so, tests for this should be.

**Review by AntonSermyazhko — DISMISSED** [review]

File: js/ui/grid_core/ui.grid_core.columns_controller.js [hunk 5]
```diff
@@ -2864,6 +2864,20 @@ module.exports = {
...
+                 * @publicName getVisibleColumnIndex(id)
+                 * @param1 id:number|string
+                 * @return number
+                 */
+                getVisibleColumnIndex: function(id, rowIndex) {
+                    let option = this.columnOption(id),
+                        index = option && option.index;
```
Comment (vconst) [inline js/ui/grid_core/ui.grid_core.columns_controller.js]: Use the next more simple code:

const index = this.columnOption(id, "index");

**Review by vconst — CHANGES_REQUESTED** [review]

File: js/ui/grid_core/ui.grid_core.columns_controller.js (hunk 5 above)
Comment (igormalzew) [reply js/ui/grid_core/ui.grid_core.columns_controller.js]: Improved column index getting

**Review by vconst — APPROVED** [review]

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**